cpu/registers.go

```diff
@@ -0,0 +1,75 @@
+package cpu
+
+// Register represents a typical 16-bit register.
+type Register uint16
+
+// Hi returns the most-significant 8 bits in a register.
+// For example, in the bit string representation of a register: 1111_1111_0000_0000
+// the first eight "1" bits would be returned as a uint8.
+func (r Register) Hi() uint8 {
+	return uint8(r >> 8)
+}
+
+// SetHi sets the most-significant 8 bits in the register.
+func (r *Register) SetHi(x uint8) {
+	*r |= Register(x) << 8
+}
```

Comment:
Notice the `(r *Register)` here. This is important, and it means value `r` is a pointer, which allows us to set its value.

Reply:
Fixed